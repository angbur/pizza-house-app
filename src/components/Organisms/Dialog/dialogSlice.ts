import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { RootState } from 'store/store';

export enum FormType {
  register = 'register',
  login = 'login',
  orderDetail = 'orderDetail',
}

type DialogState = {
  isOpen: boolean;
  formType: FormType | undefined;
};

const initialState: DialogState = {
  isOpen: false,
  formType: undefined,
};

const dialogSlice = createSlice({
  name: 'dialog',
  initialState,
  reducers: {
    openDialog: (state, action: PayloadAction<FormType>) => {
      state.isOpen = true;
      state.formType = action.payload;
    },
    closeDialog: (state) => {
      state.isOpen = false;
      state.formType = undefined;
    },
  },
});

export const { openDialog, closeDialog } = dialogSlice.actions;

export default dialogSlice.reducer;

export const selectDialogState = (state: RootState) => state.dialog;
