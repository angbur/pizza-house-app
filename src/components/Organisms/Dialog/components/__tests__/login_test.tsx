import { renderWithProviders } from 'test/test-utils';
import { getDialogElement } from '../../dialog.utils';
import { FormType } from '../../dialogSlice';

describe('Login content of dialog', () => {
  it('should renders login component correctly', () => {
    const initialState = {
      loggedIn: false,
      status: undefined,
      error: undefined,
      userId: undefined,
      login: undefined,
      token: undefined,
    };
    const handleChange = jest.fn();
    const dialog = () => (
      <>{getDialogElement(FormType.login)?.component({ handleChange: handleChange })}</>
    );

    const login = renderWithProviders(dialog(), {
      preloadedState: {
        user: initialState,
      },
    }).container;

    expect(login).toMatchSnapshot();
  });
});
