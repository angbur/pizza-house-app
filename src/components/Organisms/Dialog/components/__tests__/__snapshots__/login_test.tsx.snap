// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login content of dialog should renders login component correctly 1`] = `
<div>
  <form>
    <div
      class="sc-gswNZR tROUR"
    >
      <div
        class="sc-bcXHqe dHZfIZ"
      >
        <label
          style="color: rgb(21, 21, 21);"
        >
          Login
        </label>
        <input
          name="login"
          placeholder="Your login"
          style="color: rgb(21, 21, 21);"
          type="text"
          value=""
        />
        <p
          class="paragraph"
          style="font-family: Source Sans Pro; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(231, 76, 60);"
        />
      </div>
      <div
        class="sc-bcXHqe dHZfIZ"
      >
        <label
          style="color: rgb(21, 21, 21);"
        >
          Password
        </label>
        <input
          name="password"
          placeholder="Password"
          style="color: rgb(21, 21, 21);"
          type="password"
          value=""
        />
        <p
          class="paragraph"
          style="font-family: Source Sans Pro; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(231, 76, 60);"
        />
      </div>
    </div>
  </form>
</div>
`;
